Source/ReadyPlayerMe/Private/Storage/AvatarManifest.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright © 2021++ Ready Player Me
+
+
+#include "AvatarManifest.h"
+
+#include "ReadyPlayerMeSettings.h"
+#include "Storage/AvatarStorage.h"
+#include "Utils/AvatarManifestExtractor.h"
+
+void FAvatarManifest::AddAvatar(const FString& AvatarId)
+{
+	if (!AvatarRecords.Contains(AvatarId))
+	{
+		AvatarRecords.Add(AvatarId, FDateTime::Now().ToUnixTimestamp());
+	}
+	EnforceAvatarLimit();
+}
+
+void FAvatarManifest::BlockAvatar(const FString& AvatarId)
```

Comment:
What exactly does this blocking and unblocking stuff do? 🤔

Reply:
It's covering an edge case that has a very low possibility of happening. It would prevent an avatar from being deleted and introducing an undefined behavior if there is a parallel calls to request and delete the avatar.